### PR TITLE
[SYCLomatic] Optimize the implementation of dpct::bit_reverse

### PR DESCRIPTION
**Pull request by yihwang-nv**
Signed-off-by: Wang, Yihan <yihan.wang@intel.com>

### CODE REVIEW

**Review by wenhuiNi — APPROVED** [review]

File: clang/runtime/dpct-rt/include/util.hpp [hunk 1]
```diff
@@ -134,23 +134,48 @@ inline double cast_ints_to_double(int high32, int low32) {
...
-  T mask = 0;
-  size_t count = 4 * sizeof(T);
-  mask = ~mask >> count;
-  while (count) {
-    a = ((a & mask) << count) | ((a & ~mask) >> count);
-    count = count >> 1;
-    mask = mask ^ (mask << count);
-  }
-  return a;
+/// Copied from llvm/Support/MathExtras.h
```
Comment (tangjj11) [inline clang/runtime/dpct-rt/include/util.hpp]: please move to math.hpp

**Review by tangjj11 — CHANGES_REQUESTED** [review]

File: clang/runtime/dpct-rt/include/util.hpp [hunk 2]
```diff
@@ -134,23 +134,48 @@ inline double cast_ints_to_double(int high32, int low32) {
   return v1;
 }
 
+/// Macro compressed bit reversal table for 256 bits.
+///
+/// http://graphics.stanford.edu/~seander/bithacks.html#BitReverseTable
+static const unsigned char BitReverseTable256[256] = {
```
Comment (tangjj11) [inline clang/runtime/dpct-rt/include/util.hpp]: can move to detail?

**Review by tangjj11 — APPROVED** [review]

File: clang/runtime/dpct-rt/include/util.hpp (hunk 1 above)
Comment (yihwang-nv) [reply clang/runtime/dpct-rt/include/util.hpp]: Will it cause compatibility issues?

File: clang/runtime/dpct-rt/include/util.hpp (hunk 2 above)
Comment (yihwang-nv) [reply clang/runtime/dpct-rt/include/util.hpp]: Ok, done.

**Review by zhimingwang36 — APPROVED** [review]
LGTM

**Comment by zhimingwang36** [comment]
Close it,  As CPU side test fail.

**Pull request closed without merging**